packages/test/test-end-to-end-tests/src/test/blobsisAttached.spec.ts

```diff
@@ -0,0 +1,254 @@
+				container.attachState,
+				AttachState.Detached,
+				"container should be detached",
+			);
+			assert.strictEqual(
+				detachedDataStore.handle.isAttached,
+				false,
+				"data store handle should be detached",
+			);
+			assert.strictEqual(dds.handle.isAttached, false, "dds handle should be detached");
+			assert.strictEqual(blobHandle.isAttached, false, "blob handle should be detached");
+		};
+
+		const checkForAttachedHandles = (dds: SharedMap | SharedDirectory) => {
+			assert.strictEqual(
+				container.attachState,
+				AttachState.Attached,
+				"container should be attached",
+			);
+			assert.strictEqual(
+				detachedDataStore.handle.isAttached,
+				true,
+				"data store handle should be attached",
+			);
+			assert.strictEqual(dds.handle.isAttached, true, "dds handle should be attached");
+			assert.strictEqual(blobHandle.isAttached, true, "blob handle should be attached");
+		};
+
+		it("all detached", async function () {
+			checkForDetachedHandles(map);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after map is set in root directory", async function () {
+			detachedDataStore.root.set(mapId, map.handle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after directory is set in root directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after blob handle is set in map", async function () {
+			detachedDataStore.root.set("map", map.handle);
+			map.set("my blob", blobHandle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after blob handle is set in directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			directory.set("my blob", blobHandle);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after container is attached with map", async function () {
+			detachedDataStore.root.set("map", map.handle);
+			map.set("my blob", blobHandle);
+			const attachP = container.attach(
+				provider.driver.createCreateNewRequest(provider.documentId),
+			);
+			if (!driverSupportsBlobs(provider.driver)) {
+				return;
+			}
+			await attachP;
+			detachedBlobStorage.blobs.clear();
+			checkForAttachedHandles(map);
+		});
+
+		it("after container is attached with directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			directory.set("my blob", blobHandle);
+			const attachP = container.attach(
+				provider.driver.createCreateNewRequest(provider.documentId),
+			);
+			if (!driverSupportsBlobs(provider.driver)) {
+				return;
+			}
+			await attachP;
+			detachedBlobStorage.blobs.clear();
+			checkForAttachedHandles(directory);
+		});
+
+		it("after container is attached and dds is detached in map", async function () {
+			map.set("my blob", blobHandle);
+			const attachP = container.attach(
+				provider.driver.createCreateNewRequest(provider.documentId),
+			);
+			if (!driverSupportsBlobs(provider.driver)) {
+				return;
```

Comment:
this.skip()

Reply:
you can also put this in the before each, rather than each individual test

Reply:
i would do it very first thing in the before each, as not point in doing any work if the test can't run